### PR TITLE
NoSuperfluousPhpdocTagsFixer - Remove superfluous annotation `@abstract` and `@final`

### LINKED ISSUE #5985: Fixer proposal: remove `@final` annotation from blocks already marked as final
Example that doesn't need the final annotation anymore:

image

We could automate the removal of this. Simiarly for final methods, constants (php 8.1), etc.

### PR DESCRIPTION
**Pull request by liquid207**
Closes #5985

### CODE REVIEW

File: tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php [hunk 1]
```diff
@@ -1406,6 +1406,58 @@ class Foo {
     public function doFoo(iterable $bar, ?int $baz): ?array {}
 }',
             ],
+            'abstract annotation in function' => [
```
Comment (julienfalque) [inline tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php:1409]: Please add some tests that show the annotations are not removed when the class or the functions are not marked as final/abstract.

**Review by julienfalque — COMMENTED** [review]
Looking great!

**Comment by julienfalque** [comment]
Also, can you edit your commit message so that `@abstract` and `@final` are surrounded with `` ` ``? This would prevent people from being notified unnecessarily.

**Review by julienfalque — APPROVED** [review]

**Comment by SpacePossum** [comment]
Thanks for picking this up!

I did a little test and there are some cases that are not fixed as expected I think

```diff
[12:25 PM]-[possum@aquarium]-[~/work/PHP-CS-Fixer]-[git feature/improve-no-superfluous-phpdoc-tags-fixer] 
$ php php-cs-fixer fix test.php -vvv --dry-run --diff --rules=no_superfluous_phpdoc_tags
Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (fallback through xdebug.client_host/xdebug.client_port) :-(
PHP CS Fixer 3.2.0-DEV River by Fabien Potencier and Dariusz Ruminski
Runtime: PHP 7.4.23
Loaded config default from "/home/possum/work/PHP-CS-Fixer/.php-cs-fixer.dist.php".
Paths from configuration file have been overridden by paths provided as command arguments.
F                                                                                                                                                                                                      1 / 1 (100%)
Legend: ?-unknown, I-invalid file syntax (file ignored), S-skipped (cached or empty file), .-no changes, F-fixed, E-error
   1) PHP-CS-Fixer/test.php (no_superfluous_phpdoc_tags)
      ---------- begin diff ----------
--- /home/possum/work/PHP-CS-Fixer/test.php
+++ /home/possum/work/PHP-CS-Fixer/test.php
@@ -2,7 +2,6 @@
 final class B{}
 
 /**
- * @final
  */
 class Foo {
 }
```

**Review by SpacePossum — CHANGES_REQUESTED** [review]
please check comment and CI

**Comment by GrahamCampbell** [comment]
A breaking change, no? Shouldn't we add configuration to the fixer, off by default?

**Comment by julienfalque** [comment]
No, this is part of what the rule is intended to do, but was not implemented yet. If you want to have "no superfluous PHPDoc tags", you likely don't want to keep superfluous `@final`/`@abstract`.

**Comment by SpacePossum** [comment]
I left a PR on your fork for even more fixes, let me know what you think

**Comment by liquid207** [comment]
@SpacePossum I have added you commit to my branch.

**Comment by SpacePossum** [comment]
Thank you @liquid207.

**Pull request merged**